### PR TITLE
fix(config): don't always run configuration detector

### PR DESCRIPTION
**Pull request by sileht**
This code can be run only when the pull request is opened or
synchronized.

Depends-On: #4271
Fixes MRGFY-986

Change-Id: I43b29cae06bfc25a8d51758586e71e2307b3fcbf

### CODE REVIEW

File: mergify_engine/engine/__init__.py [hunk 1]
```diff
@@ -79,6 +79,25 @@ async def _check_configuration_changes(
         # As the PR is closed, we don't care about the config change detector.
         return False
 
+    # NOTE(sileht): This heuristic works only if _ensure_summary_on_head_sha()
+    # is called after _check_configuration_changes().
+    # If we don't have a summary yet, it means the pull request has just been
+    # open or synchronize or we never see it.
+    summary = await ctxt.get_engine_check_run(constants.SUMMARY_NAME)
+    print(summary)
```
Comment (jd) [inline mergify_engine/engine/__init__.py:87]:
```suggestion
```

File: mergify_engine/engine/__init__.py [hunk 2]
```diff
@@ -79,6 +79,25 @@ async def _check_configuration_changes(
...
+        and summary["output"]["title"]
+        != constants.CONFIGURATION_MUTIPLE_FOUND_SUMMARY_TITLE
+    ):
+        configuration_changed_check = await ctxt.get_engine_check_run(
+            constants.CONFIGURATION_CHANGED_CHECK_NAME
+        )
+        configuration_deleted_check = await ctxt.get_engine_check_run(
+            constants.CONFIGURATION_DELETED_CHECK_NAME
+        )
+        return bool(configuration_changed_check) or bool(configuration_deleted_check)
```
Comment (jd) [inline mergify_engine/engine/__init__.py:99]:
```suggestion
        configuration_changed_check = await ctxt.get_engine_check_run(
            constants.CONFIGURATION_CHANGED_CHECK_NAME
        )
        if configuration_changed_check:
            return True
        configuration_deleted_check = await ctxt.get_engine_check_run(
            constants.CONFIGURATION_DELETED_CHECK_NAME
        )
        return bool(configuration_deleted_check)
```
avoid running 2 awaits if not needed?

**Review by jd — DISMISSED** [review]

File: mergify_engine/engine/__init__.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/engine/__init__.py:99]: all checks are always retrieved.

**Review by jd — CHANGES_REQUESTED** [review]
still one comment

**Review by jd — APPROVED** [review]

**Pull request merged**